src/particles/particle_states.jl

```diff
@@ -0,0 +1,214 @@
+    return (slashed(mom) + mass * one(DiracMatrix)) / (sqrt(abs(getT(mom)) + mass))
+end
+
+function _booster_antifermion(mom::QEDbase.AbstractFourMomentum, mass::Real)
+    return (mass * one(DiracMatrix) - slashed(mom)) / (sqrt(abs(getT(mom)) + mass))
+end
+
+"""
+```julia 
+    base_state(
+        particle::AbstractParticleType,             
+        direction::ParticleDirection,               
+        momentum::QEDbase.AbstractFourMomentum,     
+        [spin_or_pol::AbstractSpinOrPolarization]   
+    )
+```
+
+Return the base state of a directed on-shell particle with a given four-momentum. For internal usage only.
+
+# Input
+
+- `particle` -- the type of the particle, e.g. [`Electron`](@ref), [`Positron`](@ref), or [`Photon`](@ref).
+- `direction` -- the direction of the particle, i.e. [`Incoming`](@ref) or [`Outgoing`](@ref).
+- `momentum` -- the four-momentum of the particle
+- `[spin_or_pol]` -- if given, the spin or polarization of the particle, e.g. [`SpinUp`](@ref)/[`SpinDown`](@ref) or [`PolarizationX`](@ref)/[`PolarizationY`](@ref).
+
+# Output
+The output type of `base_state` depends on wether the spin or polarization of the particle passed in is specified or not.
+
+If `spin_or_pol` is passed, the output of `base_state` is
+
+```julia
+base_state(::Fermion,     ::Incoming, mom, spin_or_pol) # -> BiSpinor
+base_state(::AntiFermion, ::Incoming, mom, spin_or_pol) # -> AdjointBiSpinor
+base_state(::Fermion,     ::Outgoing, mom, spin_or_pol) # -> AdjointBiSpinor
+base_state(::AntiFermion, ::Outgoing, mom, spin_or_pol) # -> BiSpinor
+base_state(::Photon,      ::Incoming, mom, spin_or_pol) # -> SLorentzVector{ComplexF64}
+base_state(::Photon,      ::Outgoing, mom, spin_or_pol) # -> SLorentzVector{ComplexF64}
+```
+
+If `spin_or_pol` is not passed to `base_state`, the output is a `StaticVector` with both spin/polarization alignments:
+
+```julia
+base_state(::Fermion,     ::Incoming, mom) # -> SVector{2,BiSpinor}
+base_state(::AntiFermion, ::Incoming, mom) # -> SVector{2,AdjointBiSpinor}
+base_state(::Fermion,     ::Outgoing, mom) # -> SVector{2,AdjointBiSpinor}
+base_state(::AntiFermion, ::Outgoing, mom) # -> SVector{2,BiSpinor}
+base_state(::Photon,      ::Incoming, mom) # -> SVector{2,SLorentzVector{ComplexF64}}
+base_state(::Photon,      ::Outgoing, mom) # -> SVector{2,SLorentzVector{ComplexF64}}
+```
+
+# Example
+
+```julia
+
+using QEDbase
+using QEDprocesses
+
+mass = 1.0                              # set electron mass to 1.0
+px,py,pz = rand(3)                      # generate random spatial components
+E = sqrt(px^2 + py^2 + pz^2 + mass^2)   # compute energy, i.e. the electron is on-shell
+mom = SFourMomentum(E, px, py, pz)      # initialize the four-momentum of the electron
+
+# compute the state of an incoming electron with spin = SpinUp
+# note: base_state is not exported!
+electron_state = QEDProcesses.base_state(Electron(), Incoming(), mom, SpinUp())
```

Comment:
Isn't `base_state()` part of QEDbase with this PR?

Reply:
Fixed in #34